pkg/transforms/filter.go

```diff
@@ -142,7 +147,11 @@ func (f *Filter) FilterByResourceName(ctx interfaces.AppFunctionContext, data in
 		for _, reading := range existingEvent.Readings {
 			readingFilteredFor := false
 			for _, name := range f.FilterValues {
-				if reading.ResourceName == name {
+				item, err := regexp.Compile(name)
+				if err != nil {
+					return false, fmt.Errorf("bad regexp in filtering on  pipeline '%s': %s", ctx.PipelineId(), err.Error())
```

Comment:
Add the regexp, i.e. value of name.
```suggestion
					return false, fmt.Errorf("bad regexp (%s) in filtering on pipeline '%s': %s", name, ctx.PipelineId(), err.Error())
```